python/shotgun_model/shotgun_model.py

```diff
@@ -841,6 +865,62 @@ def _load_external_data(self):
         """
         pass
 
+    def _get_additional_columns(self, item, is_leaf, columns):
+        """
+        Called when an item is about to be inserted into the model, to get additional items
+        to be included in the same row as the specified item. This provides an opportunity
+        for subclasses to create one or more additional columns for each item in the model.
+
+        Note that this method is always called before inserting an item, even when loading
+        from the cache. Any data that is expensive to compute or query should be added
+        to the ShotgunStandardItem in _populate_item, since column data is not cached.
+        Also note that item population methods (_populate_item, _populate_thumbnail, etc)
+        will not be called on the return columns.
+
+        This method should return a list of QStandardItems, one for each additional column.
+        The original ShotgunStandardItem is always the first item in each row and should
+        NOT be included in the returned list. Any empty value returned by this method
+        is guaranteed to be treated as an empty list (i.e. you may return None).
+
+        This method is called after _finalize_item.
+
+        :param item: :class:`~PySide.QtGui.QStandardItem` that is about to be added to the model
+        :param is_leaf: boolean that is True if the item is a leaf item
+        :param columns: list of Shotgun field names requested as the columns from _load_data
+
+        :returns: list of :class:`~PySide.QtGui.QStandardItem`
+        """
+        # default implementation will create items for the given fields from the item if it is a leaf
+        # with the display role being the string value for the field and the actual data value in
+        # SG_ASSOCIATED_FIELD_ROLE
+        items = []
+
+        if is_leaf and columns:
+            data = get_sg_data(item)
+            for column in columns:
+                # grab the value from the requested field
+                value = data.get(column)
+
+                # set the display role to the string representation of the value
+                column_item = QtGui.QStandardItem(str(value))
```

Comment:
It seems the "primary items" are having their name formatted via the `_generate_display_name()` method. Should this do the same?